sdk/search/Azure.Search.Documents/src/Indexes/FieldBuilder.cs

```diff
@@ -0,0 +1,456 @@
+                        processedTypes.Pop();
+                    }
+                }
+
+                SearchField CreateSimpleField(SearchFieldDataType SearchFieldDataType)
+                {
+                    if (prop.SerializedName is null)
+                    {
+                        // Member is unsupported or ignored.
+                        return null;
+                    }
+
+                    // Start with a SimpleField to make sure all properties default to language defaults.
+                    SearchField field = new SimpleField(prop.SerializedName, SearchFieldDataType);
+                    foreach (Attribute attribute in attributes)
+                    {
+                        switch (attribute)
+                        {
+                            case SearchableFieldAttribute searchableFieldAttribute:
+                                ((ISearchFieldAttribute)searchableFieldAttribute).SetField(field);
+                                break;
+
+                            case SimpleFieldAttribute simpleFieldAttribute:
+                                ((ISearchFieldAttribute)simpleFieldAttribute).SetField(field);
+                                break;
+
+                            default:
+                                Type attributeType = attribute.GetType();
+
+                                // Match on name to avoid dependency - don't want to force people not using
+                                // this feature to bring in the annotations component.
+                                //
+                                // Also, ignore key attributes on sub-fields.
+                                if (attributeType.FullName == "System.ComponentModel.DataAnnotations.KeyAttribute" &&
+                                    processedTypes.Count <= 1)
+                                {
+                                    field.IsKey = true;
+                                }
+                                break;
+                        }
+                    }
+
+                    return field;
+                }
+
+                ArgumentException FailOnUnknownDataType()
+                {
+                    string errorMessage =
+                        $"Property '{prop.Name}' is of type '{prop.PropertyType}', which does not map to an " +
+                        "Azure Search data type. Please use a supported data type or mark the property with [FieldBuilderIgnore] " +
+                        $"and define the field by creating a SearchField object. See {HelpLink} for more information.";
+
+                    return new ArgumentException(errorMessage, nameof(modelType))
+                    {
+                        HelpLink = HelpLink,
+                    };
+                }
+
+                IDataTypeInfo dataTypeInfo = GetDataTypeInfo(prop.PropertyType, nameProvider);
+
+                return dataTypeInfo.Match(
+                    onUnknownDataType: () => throw FailOnUnknownDataType(),
```

Comment:
Is this the core of the changes?  Was there anything else that needs extra scrutiny?

Reply:
It's the last 6 commits since you reviewed when I had it staged in my fork entirely. As of this moment:

https://github.com/Azure/azure-sdk-for-net/pull/14023/files/4b4e2bf897bb01b9b56c1aaf3d63381c34a94e53..021993263e1bfc9874e333643fa74070494c81a8